pkg/resourceinterpreter/default/native/status_type.go

```diff
@@ -0,0 +1,9 @@
+package native
```

Comment:
Thanks for your contribution! 
Here's a little problem, the new file needs to have the license added to the header, so that it passes [ci](https://github.com/karmada-io/karmada/blob/master/hack/verify-license.sh#L31). just like this:

```shell
/*
Copyright 2024 The Karmada Authors.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/
```

Reply:
added